.github/workflows/release_github.yml

```diff
@@ -4,33 +4,79 @@ on:
   workflow_dispatch:
     inputs:
       version_level:
-        description: "Semantic version level increase."
+        description: Semantic version level increase
         required: true
         type: choice
+        default: patch
         options:
           - patch
           - minor
           - major
-
+          
 permissions:
   contents: write
-  pull-requests: write
   packages: read
+  statuses: read
+  checks: read
+  pull-requests: write
+  actions: read
+  repository-projects: read
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_PAGER: cat
 
 jobs:
+  checks:
+    name: Check requirements
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Fail if main branch was selected
+        if: ${{ github.ref_name == 'main' }}
+        run: |
+          echo "Cannot release from main branch, please select valid release branch."
+          exit 1
+
+      - name: Checkout repository
+        if: ${{ github.ref_name != 'develop' }}
+        uses: actions/checkout@v4
+
+      - name: Check if PR exists
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }}
+
+      - name: Check if PR base is main
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }} --json baseRefName -q 'if .baseRefName == "main" then "PR base is main" else error("PR base is not main") end'
+        
+      - name: Check if PR is mergeable
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }} --json mergeable -q 'if .mergeable == "MERGEABLE" then "PR is mergeable" else error("PR is not mergeable") end'
+
+      - name: Check whether PR checks pass(ed)
+        # note that this assumed there are checks to be passed; this fails if there are no checks, which is no
+        # relevant to this repo
+        if: ${{ github.ref_name != 'develop' }}
+        run: |
+          gh pr checks ${{ github.ref_name }} --watch --fail-fast
+          gh pr checks ${{ github.ref_name }} --json state -q 'if . | length == 0 then "No checks found." elif map(.state == "SUCCESS") | all then "All checks passed" else error("Not all checks passed") end'            
```

Comment:
is there a way to set the if length==0 then succeed?
I think it's rare, but can imagine minor changes that trigger no actions (e.g. because actions are only triggered on certain file type changes). In that case, it seems like there are no ways to allow a release...

Reply:
Yes, this is possible (I implemented it at some point), but there are two reasons I didn't put it in the last version.

The main reason is that if you're fast enough after pushing changes to a PR, you can trigger the action before GitHub has detected that it should do checks. It will give an error that there were no checks for this branch. This happened to me once during testing the workflows.
If we would make it so it skips this when there are no checks, it could skip it unintentionally. 

The second reason is that it's quite verbose, something like (thanks to ChatGPT):

```
if [ $exit_code -ne 0 ]; then
    # Look for the specific error message
    if echo "$output" | grep -q "no checks reported on "; then
        echo "Caught the expected error: no checks reported on the branch"
        # Handle the specific error here
    else
        # Re-raise other errors
        echo "An unexpected error occurred: $output"
        exit $exit_code
    fi
else
    echo "Command succeeded: $output"
    # Handle success case here
fi
```

I don't think it's possible no check is triggered in the current repo. If there will be such a case, it is easy to add a simple check that always succeeds just to make sure that there is always a check to make this succeed.